src/util/mpir_pmi.c

```diff
@@ -415,8 +415,10 @@ int MPIR_pmi_barrier(void)
     MPIR_ERR_CHKANDJUMP1(pmi_errno != PMI2_SUCCESS, mpi_errno, MPI_ERR_OTHER,
                          "**pmi_kvsfence", "**pmi_kvsfence %d", pmi_errno);
     /* Get a non-existent key, it only returns after every process called fence */
-    int out_len;
-    PMI2_KVS_Get(pmi_jobid, PMI2_ID_NULL, "-NONEXIST-KEY", NULL, 0, &out_len);
+    if (MPIR_Process.size > 1) {
```

Comment:
Can't we skip the whole function when size is 1?

Reply:
Yes, I think that should be OK. I'll update.